models/stock_export.py

```diff
@@ -0,0 +1,282 @@
+            pallets_by_prod[quant_prods] = quants_u.mapped('package_id')
+
+        def get_prod_qty(prod, pallet):
+            contained_quants = Quant.search(
+                [('package_id', 'child_of', pallet.ids)])
+            prod_quants = contained_quants.filtered(
+                lambda x: x.product_id == prod)
+            return sum(prod_quants.mapped('quantity'))
+
+        _logger.info(_('Creating Excel file'))
+        file_name = 'warehouse_stock_{TIMESTAMP}.xls'.format(
+            TIMESTAMP=datetime.now())
+        wb = xlwt.Workbook()
+
+        def _create_sheet(sheet_name, columns):
+            sheet = wb.add_sheet(sheet_name)
+            for col, col_title in enumerate(columns):
+                sheet.write(0, col, col_title)
+            return sheet
+
+        # 1/2) 'Stock File': one entry for each (prod, pallet) pair
+
+        stock_sheet = _create_sheet('Stock File', ['Part Number',
+                                                   'Location',
+                                                   'Pallet',
+                                                   'Quantity'])
+
+        row = 0
+        for prod, pallets_by_location in pallets_by_prod_by_loc.items():
+            for location, pallets in pallets_by_location.items():
+                for pallet_by_prod in pallets:
+                    row += 1
+                    stock_sheet.write(row, 0, prod.default_code)
+                    stock_sheet.write(row, 1, location.display_name)
+                    stock_sheet.write(row, 2, pallet_by_prod.name)
+                    stock_sheet.write(row, 3, get_prod_qty(prod, pallet_by_prod))
+
+        # 2/2) 'Stock Summary': one entry for each prod
+
+        summary_sheet = _create_sheet('Stock Summary', ['Part Number',
+                                                        'Pallet Count',
+                                                        'Quantity'])
+
+        for row, (prod, prod_pallets) in enumerate(pallets_by_prod.items(), 1):
+            summary_sheet.write(row, 0, prod.default_code)
+            summary_sheet.write(row, 1, len(prod_pallets))
+            summary_sheet.write(row, 2, get_prod_qty(prod, prod_pallets))
+
+        self.__write_workbook(wb, file_name, "Stock File")
+
+    def run_movement_file_export(self):
+        '''
+        Creates movement file, in excel format, summarising the stock
+        received and sent for a given date, i.e. goods-in and goods-out
+        stock moves done, in two different tabs containing:
+            1) Goods In:  Reference, Part number, Pallet, Quantity;
+            2) Goods Out: Reference, Part number, Pallet, Quantity.
+        '''
+        if not self.date:
+            raise ValidationError(_("Date not found."))
+
+        self.field_data = False
+        self.field_name = False
+
+        Move = self.env['stock.move']
+        picking_type_in = self.env.ref('stock.picking_type_in')
+        picking_type_out = self.env.ref('stock.picking_type_out')
```

Comment:
These picking types need to be accessed via the logged in users current warehouse (Users.get_user_warehouse())

Reply:
Thanks, will change.